### DISCUSSION TITLE
Feature Request - Bar gauge improvements? Dynamic max per bar or different query for value labels or something?

### DISCUSSION THREAD

**Discussion by DoTheEvo** [grafana/grafana#66159] [Dashboards, Panels & Visualizations]
First what should the outcome look like.
Look how great Bar gauge example looks in the official documentation.

![gauge_bar](https://i.imgur.com/4YAKGDH.png)

But while this is nice, its kinda cheating as nobody really can use it as shown. Not unless they have all disks same size.

So would it not be great to be able to use grafana to show stuff like in the example, but with disks that can range from 32GB to 24TB while maintaining the look and showing GB used?

That is all.

----

Now my spitballing take.

* obviously the main query showing the fullness like in the pic needs to be in the end about percent, and that is no issue currently, I have it done like that.
* but then the example pic shows also GB info, data used and it is kinda important, cuz free space could be 2GB or 2TB and you dunno that from percent view. To get that info in to the graph that would require a different query to be in value label. And the ability to set value displayed to that query rather than what generated bar graph
* or percents would not be used and it would just need dynamic max set, that kinda is in beta - *Config from query results* , but if I am not mistaken its the max for the whole bar line, not for individual bars, they would need to be discrete separated from each other. Theres also kinda issue that I got a feeling from *Config from query results* that its not universal, that I can do it only per returned item in a query results but I dunno. Solution should be universal that when in a month new machine starts reporting disk use its automatically there without work.

I dont really know.
I just know I would really want whats shown in the example pic and I think people would use it often.
And grafana team figured out how to do other very complicated stuff so I hope this could be done too.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82757. Feel free to continue the discussion around this there. Thank you!
